lib/jekyll_pages_api/page.rb

```diff
@@ -18,23 +21,31 @@ def filterer
     end
 
     def title
-      self.filterer.decode_html(self.page.data['title'] || '')
+      title = self.page.data['title'] if self.page.respond_to?(:data)
+      title = self.page.title if title == nil and self.page.respond_to?(:title)
+      self.filterer.decode_html(title || '')
     end
 
     def base_url
-      self.page.site.baseurl
+      @site.baseurl
     end
 
     def url
-      [self.base_url, self.page.url].join
+      rel_path = self.page.url if self.page.respond_to?(:url)
+      if rel_path == nil and self.page.respond_to?(:relative_path)
+        rel_path = self.page.relative_path
+      end
+      [self.base_url, rel_path].join
     end
 
     def body_text
-      self.filterer.text_only(self.page.content)
+      output = self.page.content if self.page.respond_to?(:content)
+      File.open(self.page.path, 'r') {|f| output = f.read} if output == nil
```

Comment:
Dammit, I keep expecting http://ruby-doc.org/core-2.2.0/ to have the _complete_ documentation; now I know to always check `ri`. Done.